### PR TITLE
feat : changes for merging our ingestion into one

### PR DESCRIPTION
**Pull request by kotharironak**
- merges span-normalizer + raw-spans-grouper into hypertrace-ingester
- needs some config related changes in the framework for this PR to work
- for now, it's more on iteration with Laxman and Suarj

### CODE REVIEW

File: hypertrace-ingester/src/main/java/org/hypertrace/ingester/HypertraceIngester.java [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+  @Override
+  public StreamsBuilder buildTopology(Map<String, Object> streamsProperties,
+      StreamsBuilder streamsBuilder,
+      Map<String, KStream<?, ?>> inputStreams) {
+
+    // create span normalizer
+    SpanNormalizer spanNormalizer = new SpanNormalizer(ConfigClientFactory.getClient());
+    Config spanNormalizerConfig = getJobConfig("span-normalizer");
+    Map<String, Object> spanNormalizerConfigMap = spanNormalizer.getStreamsConfig(spanNormalizerConfig);
+    streamsBuilder = spanNormalizer.buildTopology(spanNormalizerConfigMap, streamsBuilder, inputStreams);
```
Comment (kotharironak) [inline hypertrace-ingester/src/main/java/org/hypertrace/ingester/HypertraceIngester.java:44]: I have some local changes in spanNomralizer to get topics from getAppConfig() related. Something like as below

```
private Config getConfig() {
    if (jobConfig != null) return jobConfig;
    else return getAppConfig();
  }
```

File: hypertrace-ingester/src/main/java/org/hypertrace/ingester/HypertraceIngester.java [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+    // create raw spans grouper
+    RawSpansGrouper rawSpansGrouper = new RawSpansGrouper(ConfigClientFactory.getClient());
+    Config rawSpansGrouperConfig = getJobConfig("raw-spans-grouper");
+    Map<String, Object> rawSpansGrouperConfigMap = rawSpansGrouper.getStreamsConfig(rawSpansGrouperConfig);
+    streamsBuilder = rawSpansGrouper.buildTopology(rawSpansGrouperConfigMap, streamsBuilder, inputStreams);
```
Comment (kotharironak) [inline hypertrace-ingester/src/main/java/org/hypertrace/ingester/HypertraceIngester.java:50]: I have some local changes in spanNomralizer to get topics from getAppConfig() related. Something like as below
```
private Config getConfig() {
    if (jobConfig != null) return jobConfig;
    else return getAppConfig();
  }
```

**Comment by kotharironak** [comment]
- updated with `hypertrace-trace-enricher`

**Review by jcchavezs — DISMISSED** [review]
LGTM

**Review by JBAhire — APPROVED** [review]

File: .circleci/config.yml [hunk 3]
```diff
@@ -63,7 +63,7 @@ jobs:
     executor: gradle_docker
     steps:
       - setup_build_environment
-      - setup_remote_docker:
+      - setup_remote_docker: &latest_remote_docker
           version: 19.03.12
```
Comment (laxmanchekka) [inline .circleci/config.yml:67]: Should we remove this value?

File: .gitmodules [hunk 4]
```diff
@@ -0,0 +1,16 @@
+[submodule "span-normalizer"]
+	path = span-normalizer
+	url = https://github.com/hypertrace/span-normalizer.git
+	branch = main
```
Comment (laxmanchekka) [inline .gitmodules:4]: I think its better to point to a released commit for better stability.

Comment (jcchavezs) [reply .gitmodules:4]: I might be wrong but those modules are set in a commit unless we do git submodule update isn't? 
image

File: .circleci/config.yml (hunk 3 above)
Comment (adriancole) [reply .circleci/config.yml:67]: probably instead add the missing comment

```
      # The remote docker version is independent from what's installed in the gradle_docker image and defaults to Docker 17.
      #   Use the latest value from https://circleci.com/docs/2.0/building-docker-images/#docker-version
```

File: .gitmodules (hunk 4 above)
Comment (adriancole) [reply .gitmodules:4]: https://github.com/hypertrace/hypertrace-ingester/issues/8

**Pull request merged**